common/common.go

```diff
@@ -51,10 +51,18 @@ func initialiseHTTPClient() {
 	}
 }
 
-// NewHTTPClientWithTimeout initialises a new HTTP client with the specified
-// timeout duration
+// NewHTTPClientWithTimeout initialises a new HTTP client and its underlying
+// transport IdleConnTimeout with the specified timeout duration
 func NewHTTPClientWithTimeout(t time.Duration) *http.Client {
-	h := &http.Client{Timeout: t}
+	tr := &http.Transport{
```

Comment:
There's 2 things left over to contend with:

1) If someone uses a proxy for an exchange, the transport settings here will be overwritten, making this ineffective
2) When an exchange is loaded in SetupDefaults, I think we should also upgrade the SetHTTPClientTimeout method to also adjust the transport IdleConnTimeout too so that both are in line with each other